crypto/provider_conf.c

```diff
@@ -275,10 +344,14 @@ static int provider_conf_load(OSSL_LIB_CTX *libctx, const char *name,
     }
 
     /*
-     * Even if ok is 0, we still return success. Failure to load a provider is
-     * not fatal. We want to continue to load the rest of the config file.
+     * Provider activation returns a tristate:
```

Comment:
Above on line 337 provider_conf_params() is called which can return -1 but it is not handled correctly in the statements after that.

Reply:
fixed